sweepbatcher/sweep_batcher_test.go

```diff
@@ -1944,6 +1952,74 @@ func testSweepBatcherCloseDuringAdding(t *testing.T, store testStore,
 	<-registrationChan
 }
 
+// testCustomSignMuSig2 tests the operation with custom musig2 signer.
+func testCustomSignMuSig2(t *testing.T, store testStore,
+	batcherStore testBatcherStore) {
+
+	defer test.Guard(t)()
+
+	lnd := test.NewMockLnd()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	sweepStore, err := NewSweepFetcherFromSwapStore(store, lnd.ChainParams)
+	require.NoError(t, err)
+
+	// Use custom MuSig2 signer function.
+	batcher := NewBatcher(lnd.WalletKit, lnd.ChainNotifier, lnd.Signer,
+		nil, nil, lnd.ChainParams, batcherStore,
+		sweepStore, WithSignMuSig2(testSignMuSig2func))
```

Comment:
I'd also check that the signature is as expected so we test that the custom signer was indeed used.

Reply:
Fixed